### DISCUSSION TITLE
C# / ETO Forms / .NET 7 on Windows machine for deployment to macOS?

### DISCUSSION THREAD

**Discussion by git-hub-9999** [picoe/Eto#2470] [Q&A]
I am hoping to compile a c# desktop project on a Windows machine (with ETO forms as the GUI), then run that resulting program on a Mac (without using Mono).

Is that theoretically possible using .NET 7 (and ETO Forms GUI)? Or do I have to re-compile on a Mac machine (via Visual Studio for Mac)?

I tried playing with the ETO forms template sample but it gives errors (can give more detail on that if required).

Previously I was using:
.net framework / ETO Forms / Mono and I was able to successfully compile Windows, Mac and Linux versions from the same Windows machine. My understanding is that Mono is not required for .NET 7 which is fantastic but perhaps I can no longer compile Mac deployments from Windows?

Thank you very much for all your work to make this library happen. It has been extremely useful considering that Microsoft hasn't provided an adequate solution for this yet.

**Comment by cwensley** [comment]
The quick answer is: nope.  

The problem being is that Windows does not have any concept of executable files, and the second problem is the issue of [signing and notarization](https://github.com/picoe/Eto/wiki/Mac-Code-Signing-and-Notarization), which can only be done on a Mac.

If you don't sign/notarize the user must [follow these steps](https://github.com/picoe/Eto/wiki/Troubleshooting-and-Common-Problems#problem-the-app-bundle-application-compiled-on-windows-does-not-run-on-mac) after downloading and unzipping the .app bundle which is not user friendly.

Note that you can only build the Eto.Mac64 platform on Windows and Linux, the Eto.MacOS platform does not work as it relies on Microsoft's tooling which only works on Mac.

Hope this helps!
